win32ss/user/ntuser/caret.c

```diff
@@ -41,6 +43,13 @@ co_IntDrawCaret(PWND pWnd, PTHRDCARETINFO CaretInfo)
        NtGdiSaveDC(hdc);
     }
 
+    IntGetClientRect(pWnd, &rcClient);
+    NtGdiIntersectClipRect(hdc,
```

Comment:
There is still no explanation, why this is required. The returned DC is an owned/class DC (as opposed to a cache DC), i.e. it is persistent and should never change under the hood. This code changes the DC's clipping region. Instead my suggestion is to use DCX_CACHE, which gives a fresh DC from the cache and no intersect and no save/restore should be required.